.github/workflows/build-test-docker.yaml

```diff
@@ -83,3 +102,66 @@ jobs:
         with:
           name: ${{ inputs.artifact-name }}
           path: /tmp/image.tar
+  build-test-docker-multi-arch:
+    name: Build and test multi-arch Semgrep Docker image
+    if: ${{ inputs.multi-arch }}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+    strategy:
+      matrix:
+        architecture:
+          - amd64
+          - arm64
+    steps:
+      - if: ${{ matrix.architecture != 'amd64' }}
+        uses: docker/setup-qemu-action@v2
+      - uses: docker/setup-buildx-action@v2
+      - id: meta
+        name: Set tags and labels
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ inputs.repository-name }}
+          # :latest is also added automatically
+          tags: ${{ inputs.docker-tags }}
+      - uses: depot/setup-action@v1
+        if: ${{ inputs.use-depot-for-docker-build }}
+      - name: Build image (via depot.dev)
+        id: build-image-depot
+        if: ${{ inputs.use-depot-for-docker-build }}
+        uses: depot/build-push-action@v1.7.1
+        with:
+          project: fhmxj6w9z8
```

Comment:
How does authentication with depot work? Can we limit execution to only workflows running in this repository (i.e. not forks or other repos where this project id is specified in a workflow)?

Reply:
tl;dr: from offline conversation, yes - this is handled by OIDC.